internal/infraenv/infraenv.go

```diff
@@ -31,6 +34,35 @@ func NewManager(log logrus.FieldLogger, db *gorm.DB, objectHandler s3wrapper.API
 	}
 }
 
+func (m Manager) DeleteOrphanInfraEnvs(ctx context.Context, inactiveSince strfmt.DateTime) error {
+	log := logutil.FromContext(ctx, m.log)
+	var infraEnvs []*models.InfraEnv
+
+	if err := m.db.Where("updated_at < ?", inactiveSince).Find(&infraEnvs).Error; err != nil {
+		return err
+	}
+	for _, infraEnv := range infraEnvs {
+		if infraEnv.ClusterID != "" {
```

Comment:
I think that you can delete the InfraEnv if `ClusterID` is empty. after all this infra env is old and unused.

Reply:
Done